assets/source/setup-guide/app/helpers/connect-advertiser.js

```diff
@@ -0,0 +1,27 @@
+/**
+ * External dependencies
+ */
+import apiFetch from '@wordpress/api-fetch';
+
+const connectAdvertiser = async ( trackingAdvertiser, trackingTag ) => {
+	try {
+		const results = await apiFetch( {
+			path: `${ wcSettings.pinterest_for_woocommerce.apiRoute }/tagowner/`,
+			data: {
+				advrtsr_id: trackingAdvertiser,
+				tag_id: trackingTag,
+			},
+			method: 'POST',
+		} );
+
+		if ( trackingAdvertiser === results.connected ) {
+			return true;
+		}
+
+		return false;
```

Comment:
Based on your comment in https://github.com/woocommerce/pinterest-for-woocommerce/pull/305#discussion_r780374933:

> The boolean true/false indicates if the account was reconnected (reconnected means that a new advertiser or tag is connected to the platform). If the advertiser and tag were already connected (no changes in the previous values), `connectAdvertiser` will return false.

I think this line should be like the following instead?

```suggestion
		return results.reconnected;
```

`results.connected` is the account ID, and `results.reconnected` is boolean true/false after changing advertiser and tracking tag.

![image](https://user-images.githubusercontent.com/417342/148742382-7bf74102-fa76-4170-ace5-ddffa304c616.png)

📹 Here's a voice-annotated demo video (based on the current code [`3c82593` (#305)](https://github.com/woocommerce/pinterest-for-woocommerce/pull/305/commits/3c825936a6dbc7c66e0adacd0488c9344c766c8d), not the suggested change) showing that the notice `'The advertiser was connected successfully.'` will always show even if we did not change the advertiser and tracking tag.

https://user-images.githubusercontent.com/417342/148743590-99a0d7a6-441a-4dc4-8bd7-294349d68ab9.mov

Reply:
Good caught @ecgan. Correct, we need to return the reconnected parameter in the request 👍